### ISSUE TITLE
Feature problems with characters (ࢺ ࢹ ࢸ ࢷ ࢶ)

### ISSUE THREAD

**Issue by ibanaham** [harfbuzz/harfbuzz#2266]
Issues with the following characters:

ࢺ ࢹ ࢸ ࢷ ࢶ

Their open type features are not recognised by Adobe or Microsoft applications (same problem with Android). Their initial, medial and final forms do not work but only their isolated forms appear. 

Please help, thanks.

**Comment by jfkthame** [comment]
Adobe and Microsoft applications are the responsibility of those companies.

These characters should work fine in HarfBuzz provided the font being used has support for them. What font are you using on Android?

**Comment by ibanaham** [comment]
I am using a default font on my android phone. I do not have the option of changing it unfortunately. I tried various applications these are my results: 

https://www.docdroid.net/vNvnAGN/problems-with-characters.pdf

I was using a font that I adapted just to run some checks with. Do you know when Adobe and Microsoft will be using HarfBuzz. Recently I received a message from a member of the adobe team saying: “We are aware of the Arabic shaping issues in our apps and are in the process of updating them to use HarfBuzz to resolve those issues.”

**Comment by behdad** [comment]
Sounds like an itemizer issue to me.  I'll read and comment more.

**Comment by jfkthame** [comment]
>  I tried various applications these are my results:

Looks like harfbuzz shapes these characters fine, as seen in LibreOffice. (I don't know what shaping engine WhatsApp PC is using.)

On Android, your font just doesn't support the characters at all -- nothing we can do about that.

Acrobat isn't responsible for shaping, it just displays whatever glyphs are in the PDF, so that depends on the PDF-generating application.

That leaves MS Office and Adobe Illustrator. As far as I know, neither of them is using harfbuzz.

So in short, I don't see any evidence here of a *harfbuzz* problem.

And no, we can't tell you if or when various companies will fix their products.

**Comment by ibanaham** [comment]
I understand. Well thank you for taking the time to respond to me. I hope that Harfbuzz will be used by Adobe and Microsoft someday.

**Comment by ebraminio** [comment]
Thanks :)

**Comment by behdad** [comment]
Adobe Illustrator 2018 and Adobe Photoshop 2018 and later should support HarfBuzz.  You would need to manually pick the "world composer" or something to that effect though.

**Comment by ibanaham** [comment]
Hi Behdad. I tried those letters (ࢺ ࢹ ࢸ ࢷ ࢶ) using Adobe Illustrator 2020 and unfortunately they did not work although other Arabic letters join seamlessly. I will email you a font so you can try it out. 

Thanks

**Comment by musharm** [comment]
Thanks for reporting the issue. Adobe products are currently using Harfbuzz for some languages in the World Ready Composer. Arabic and other RTL scripts are in our plan too.

-Mukul
